<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CMvChipMainGrid</class>
 <widget class="QWidget" name="CMvChipMainGrid">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>892</width>
    <height>652</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>晶片主栅线</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_3">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <property name="spacing">
      <number>0</number>
     </property>
     <item>
      <widget class="QGraphicsView" name="graphicsView_mainImage">
       <property name="minimumSize">
        <size>
         <width>500</width>
         <height>600</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>16777215</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QFrame" name="frame">
       <property name="minimumSize">
        <size>
         <width>500</width>
         <height>42</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>500</width>
         <height>42</height>
        </size>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>2</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QPushButton" name="pushButton_zoomIn">
          <property name="minimumSize">
           <size>
            <width>70</width>
            <height>40</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>70</width>
            <height>40</height>
           </size>
          </property>
          <property name="text">
           <string>放大</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_zoomOut">
          <property name="minimumSize">
           <size>
            <width>70</width>
            <height>40</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>70</width>
            <height>40</height>
           </size>
          </property>
          <property name="text">
           <string>缩小</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_zoomBest">
          <property name="minimumSize">
           <size>
            <width>70</width>
            <height>40</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>70</width>
            <height>40</height>
           </size>
          </property>
          <property name="text">
           <string>最好尺寸</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_lockRoiMove">
          <property name="minimumSize">
           <size>
            <width>70</width>
            <height>40</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>70</width>
            <height>40</height>
           </size>
          </property>
          <property name="text">
           <string>锁定ROI</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>63</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="label_mousePos">
          <property name="minimumSize">
           <size>
            <width>150</width>
            <height>40</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>150</width>
            <height>40</height>
           </size>
          </property>
          <property name="text">
           <string>TextLabel</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="spacing">
      <number>0</number>
     </property>
     <item>
      <widget class="QTabWidget" name="tabWidget">
       <property name="minimumSize">
        <size>
         <width>382</width>
         <height>602</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="tab">
        <attribute name="title">
         <string>输入设置</string>
        </attribute>
        <widget class="QLabel" name="label_funcName">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>30</y>
           <width>81</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>检测器名称：</string>
         </property>
        </widget>
        <widget class="QCheckBox" name="checkBox_enableFunc">
         <property name="geometry">
          <rect>
           <x>270</x>
           <y>20</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <property name="minimumSize">
          <size>
           <width>101</width>
           <height>31</height>
          </size>
         </property>
         <property name="text">
          <string>启用检测器</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QPlainTextEdit" name="plainTextEdit_funcName">
         <property name="geometry">
          <rect>
           <x>130</x>
           <y>20</y>
           <width>131</width>
           <height>31</height>
          </rect>
         </property>
         <property name="minimumSize">
          <size>
           <width>131</width>
           <height>31</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>131</width>
           <height>31</height>
          </size>
         </property>
         <property name="overwriteMode">
          <bool>false</bool>
         </property>
        </widget>
        <widget class="QTableWidget" name="tableWidget_input">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>70</y>
           <width>341</width>
           <height>391</height>
          </rect>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="textElideMode">
          <enum>Qt::ElideRight</enum>
         </property>
         <property name="verticalScrollMode">
          <enum>QAbstractItemView::ScrollPerItem</enum>
         </property>
         <property name="showGrid">
          <bool>false</bool>
         </property>
         <attribute name="horizontalHeaderStretchLastSection">
          <bool>true</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderCascadingSectionResizes">
          <bool>false</bool>
         </attribute>
         <row>
          <property name="text">
           <string>1</string>
          </property>
         </row>
         <column>
          <property name="text">
           <string>输入名称</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>输入来源</string>
          </property>
         </column>
         <item row="0" column="0">
          <property name="text">
           <string>黑白图像</string>
          </property>
         </item>
         <item row="0" column="1">
          <property name="text">
           <string>某个检测器的输出</string>
          </property>
         </item>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_2">
        <attribute name="title">
         <string>参数设置</string>
        </attribute>
        <widget class="QComboBox" name="comboBox_binaryDetection">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>50</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <item>
          <property name="text">
           <string>手动白色</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>手动黑色</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>自动白色</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>自动黑色</string>
          </property>
         </item>
        </widget>
        <widget class="QLabel" name="label_binaryDetection">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>20</y>
           <width>111</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>二值化类型：</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_grayLevel">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>130</x>
           <y>20</y>
           <width>111</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>晶片灰度阈值：</string>
         </property>
        </widget>
        <widget class="QSpinBox" name="spinBox_grayLevel">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>130</x>
           <y>50</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <property name="maximum">
          <number>255</number>
         </property>
        </widget>
        <widget class="QCheckBox" name="checkBox_imagePreprocessing">
         <property name="geometry">
          <rect>
           <x>250</x>
           <y>110</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <property name="minimumSize">
          <size>
           <width>101</width>
           <height>31</height>
          </size>
         </property>
         <property name="text">
          <string>开启</string>
         </property>
        </widget>
        <widget class="QSpinBox" name="spinBox_branchesNumber">
         <property name="geometry">
          <rect>
           <x>250</x>
           <y>50</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <property name="maximum">
          <number>5000</number>
         </property>
        </widget>
        <widget class="QLabel" name="label_branchesNumber">
         <property name="geometry">
          <rect>
           <x>250</x>
           <y>20</y>
           <width>111</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>主栅线条数：</string>
         </property>
        </widget>
        <widget class="QSpinBox" name="spinBox_areaMax">
         <property name="geometry">
          <rect>
           <x>130</x>
           <y>230</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <property name="maximum">
          <number>5000000</number>
         </property>
        </widget>
        <widget class="QLabel" name="label_areaMin">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>200</y>
           <width>111</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>每条栅线面积下限：</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_areaMax">
         <property name="geometry">
          <rect>
           <x>130</x>
           <y>200</y>
           <width>111</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>每条栅线面积上限：</string>
         </property>
        </widget>
        <widget class="QSpinBox" name="spinBox_areaMin">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>230</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <property name="maximum">
          <number>5000000</number>
         </property>
        </widget>
        <widget class="QGroupBox" name="groupBox_imagePreprocessing">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>100</y>
           <width>231</width>
           <height>91</height>
          </rect>
         </property>
         <property name="title">
          <string>图像预处理</string>
         </property>
         <widget class="QCheckBox" name="checkBox_displayImage">
          <property name="geometry">
           <rect>
            <x>120</x>
            <y>50</y>
            <width>101</width>
            <height>31</height>
           </rect>
          </property>
          <property name="minimumSize">
           <size>
            <width>101</width>
            <height>31</height>
           </size>
          </property>
          <property name="text">
           <string>显示预处理图像</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_corrosiveCore">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>20</y>
            <width>111</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>腐蚀内核：</string>
          </property>
         </widget>
         <widget class="QComboBox" name="comboBox_corrosiveCore">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>50</y>
            <width>101</width>
            <height>31</height>
           </rect>
          </property>
          <item>
           <property name="text">
            <string>3x3</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>5x5</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>7x7</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>9x9</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>11x11</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>13x13</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>15x15</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>17x17</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>19x19</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>21x21</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>23x23</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>25x25</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>27x27</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>29x29</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>31x31</string>
           </property>
          </item>
         </widget>
         <widget class="QLabel" name="label_displayImage">
          <property name="geometry">
           <rect>
            <x>120</x>
            <y>20</y>
            <width>111</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>图像显示:</string>
          </property>
         </widget>
        </widget>
        <widget class="QPushButton" name="pushButton_initialize">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>300</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>学习当前值</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_initialize">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>270</y>
           <width>111</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>栅线初始位置设定：</string>
         </property>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_4">
        <attribute name="title">
         <string>结果绘制</string>
        </attribute>
        <widget class="QCheckBox" name="checkBox_valuTrig">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>20</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>启用绘制</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </widget>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="leftMargin">
        <number>1</number>
       </property>
       <property name="topMargin">
        <number>1</number>
       </property>
       <property name="rightMargin">
        <number>1</number>
       </property>
       <property name="bottomMargin">
        <number>1</number>
       </property>
       <item>
        <widget class="QPushButton" name="pushButton_ones">
         <property name="minimumSize">
          <size>
           <width>80</width>
           <height>40</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>40</height>
          </size>
         </property>
         <property name="text">
          <string>单次</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_affirm">
         <property name="minimumSize">
          <size>
           <width>80</width>
           <height>40</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>40</height>
          </size>
         </property>
         <property name="text">
          <string>确定</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pbPrev_cancel">
         <property name="minimumSize">
          <size>
           <width>80</width>
           <height>40</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>40</height>
          </size>
         </property>
         <property name="text">
          <string>取消</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="CMvContourMatching.qrc"/>
 </resources>
 <connections/>
</ui>
